src/engine/Engine.cpp

```diff
@@ -1811,6 +1811,15 @@ bool Engine::shouldExitDueToTimeout( unsigned timeout ) const
     return _statistics.getTotalTime() / MILLISECONDS_TO_SECONDS > timeout;
 }
 
+void Engine::reset()
+{
+    Statistics *statistics = new Statistics();
+    resetStatistics( *statistics );
+    clearViolatedPLConstraints();
+    resetSmtCore();
+    resetBoundTighteners();
+    resetExitCode();
+}
 
 void Engine::resetStatistics( const Statistics &statistics )
 {
```

Comment:
I noticed that we're setting the active entry strategy in this method. This doesn't seem to belong here.

Reply:
Ok, I removed that line